.envrc

```diff
@@ -85,68 +95,79 @@ export PIP_DISABLE_PIP_VERSION_CHECK=on
 # increase node's memory limit, required for our webpacking
 export NODE_OPTIONS=--max-old-space-size=4096
 
-
 ### System ###
 
 for pkg in \
-    make            \
-    docker          \
-    chromedriver    \
-    pkg-config      \
-    openssl         ;
-    do
+    make \
+    docker \
+    chromedriver \
+    pkg-config \
+    openssl; do
     if ! require "$pkg"; then
         die "You seem to be missing the system dependency: ${pkg}
 Please install homebrew, and run brew bundle."
     fi
 done
 
-
 ### Git ###
 
 info "Configuring git..."
 
 make setup-git-config
 
-
 ### Python ###
 
+# we're enforcing that virtualenv be in .venv{,3}, since future tooling e.g.
+# venv-update will rely on this.
+venv_name=".venv"
+python_version="python2.7"
+
+if [ "x$SENTRY_PYTHON3" = "x1" ]; then
+    venv_name=".venv3"
+    python_version="python3.6"
+
+    warn "Running in EXPERIMENTAL Python 3 mode... Good luck"
+fi
+
 info "Activating virtualenv..."
 
-# we're enforcing that virtualenv be in .venv, since future tooling e.g. venv-update will rely on this.
-if [ ! -f ".venv/bin/activate" ]; then
+if [ ! -f "${venv_name}/bin/activate" ]; then
     info "You don't seem to have a virtualenv."
-    advice_init_venv
+    advice_init_venv "$python_version" "$venv_name"
 fi
 
 # The user might be cd'ing into sentry with another non-direnv managed
 # (in that it would be automatically deactivated) virtualenv active.
 deactivate 2>/dev/null || true
 
-source .venv/bin/activate
+source ${venv_name}/bin/activate
 
 # XXX: ideally, direnv is able to export PS1 as modified by sourcing venvs
 #      but we'd have to patch direnv, and ".venv" isn't descriptive anyways
 unset PS1
 
 # We're explicitly disallowing symlinked venvs (python would resolve to the canonical location)
-if [ "$(command -v python)" != "${PWD}/.venv/bin/python" ]; then
+if [ "$(command -v python)" != "${PWD}/${venv_name}/bin/python" ]; then
     info "Failed to activate virtualenv. Your virtualenv's probably symlinked." \
-    "We want everyone to be on the same page here, so you'll have to recreate your virtualenv."
-    advice_init_venv
+        "We want everyone to be on the same page here, so you'll have to recreate your virtualenv."
+    advice_init_venv "$python_version" "$venv_name"
 fi
 
-python -c "import sys; sys.exit(sys.version_info[:2] != (2, 7))" || \
-    die "For some reason, the virtualenv isn't Python 2.7."
+if [ "x$SENTRY_PYTHON3"  = "x1" ]; then
```

Comment:
remove the x idk why people do that lol

Reply:
Posterity: apparently really old shells aren't good with empty strings. This isn't a concern in 2020 though. Well. One can only hope.